lib/src/license_detection/confidence.dart

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+import 'package:pana/src/third_party/diff_match_patch/diff.dart';
+import 'package:pana/src/license_detection/license.dart';
+import 'package:pana/src/license_detection/token_matcher.dart';
+import 'package:pana/src/license_detection/tokenizer.dart';
+
+/// Computes the confidence of [knownLicense] matching with [unknownLicense] and
+/// returns an instance of [LicenseMatch] or null based on it.
+LicenseMatch? confidenceMatch(
+  LicenseWithNGrams unknownLicense,
+  LicenseWithNGrams knownLicense,
+  MatchRange matchRange,
+  double threshold,
+) {
+  final diffs = getDiffs(
+    unknownLicense.tokens,
+    knownLicense.tokens,
+    matchRange,
+  );
+
+  final range = diffRange(tokensNormalizedValue(knownLicense.tokens), diffs);
+  final distance =
+      scoreDiffs(diffs.skip(range.start).take(range.end - range.start));
```

Comment:
Why don't we just do:
```suggestion
  diffs = List.unmodifiable(diffs.skip(range.start).take(range.end - range.start).toList());
  final distance = scoreDiffs(diffs);
```

That way we never have to think about `range` again, right?

The entries in `diffs` before `range.start` should they ever be used for anything?

The resulting `LicenseMatch` would also be simpler if we just removed the `diffs` we don't care about.

Can we do that? does that make sense?